### PR TITLE
[CLS-10991] Natis token secret

### PR DESCRIPTION
**Pull request by natishauli**

### CODE REVIEW

File: charts/s1-agent/templates/_helpers.tpl [hunk 1]
```diff
@@ -187,6 +187,26 @@ Generate certificates for helper secret
 {{- dict "tls.crt" $tlsCert "tls.key" $tlsKey "ca.crt" $caCert | toYaml -}}
 {{- end -}}
 
+{{- define "helper_token.secret.create" -}}
+{{- empty .Values.secrets.helper_token | ternary "true" "" }}
+{{- end -}}
+
+{{- define "helper_token.secret.name" -}}
+{{- if include "helper_token.secret.create" . }}
+{{- include "helper.fullname" . -}}
```
Comment (danisevas) [inline charts/s1-agent/templates/_helpers.tpl:196]: is this needed?

File: charts/s1-agent/templates/_helpers.tpl [hunk 2]
```diff
@@ -187,6 +187,26 @@ Generate certificates for helper secret
 {{- dict "tls.crt" $tlsCert "tls.key" $tlsKey "ca.crt" $caCert | toYaml -}}
 {{- end -}}
 
+{{- define "helper_token.secret.create" -}}
+{{- empty .Values.secrets.helper_token | ternary "true" "" }}
```
Comment (danisevas) [inline charts/s1-agent/templates/_helpers.tpl:191]: do we need the ternary? because `empty .Values.secrets.helper_token` will return bool

**Review by danisevas — CHANGES_REQUESTED** [review]

File: charts/s1-agent/templates/_helpers.tpl (hunk 1 above)
Comment (natishauli) [reply charts/s1-agent/templates/_helpers.tpl:196]: As same as site key and other secrets, we allow customers to set their own token secret and just set the name. So in that case we don't create a token secret, just using the one that was set by the customer.

File: charts/s1-agent/templates/common/configmaps.yaml [hunk 3]
```diff
@@ -55,4 +63,13 @@ data:
...
+        - name: S1_AGENT_CONFIG_FILE
+          value: "/{{ include "agentInjection.name" . }}/config.yaml"
 {{ end }}
+        volumeMounts:
+        - name: {{ include "agentInjection.name" . }}-config
+          mountPath: /{{ include "agentInjection.name" . }}
+      volumes:
+        - name: {{ include "agentInjection.name" . }}-config
+          configMap:
+            name: {{ include "agentInjection.name" . }}
```
[the remark this answers is not in the document]
Comment (natishauli) [reply charts/s1-agent/templates/common/configmaps.yaml:75]: It was supposed to be part of the uninstall PR and somehow got to here. I created a different PR for it and removed it from this PR.

File: charts/s1-agent/values.yaml [hunk 4]
```diff
@@ -45,6 +45,7 @@ configuration:
 secrets:
   imagePullSecret: "" # you need to specify the name of the image pull secret (created outside this chart)
   helper_certificate: "" # you need to specify the name of the helper signed certificate secret (created outside this chart)
+  helper_token: "" # you need to specify the name of the helper token secret (created outside this chart)
```
[the remark this answers is not in the document]
Comment (natishauli) [reply charts/s1-agent/values.yaml:48]: yes it is, unless a customer wants to use his own token secret, same as certificate.

File: charts/s1-agent/templates/_helpers.tpl (hunk 2 above)
Comment (natishauli) [reply charts/s1-agent/templates/_helpers.tpl:191]: yes, if helper token secret name was not set by a customer, we should create it.

**Review by danisevas — APPROVED** [review]

**Pull request merged**